src/GameLogic/DataStructures/ActionSequence.ts

```diff
@@ -13,8 +13,9 @@ export class ActionSequence {
 		return this._actionSequence;
 	}
 
-	constructor(actionSequence: PlayerAction[] = []) {
+	constructor(actionSequence: PlayerAction[] = [], position = 0) {
```

Comment:
I am not sure if I am happy with passing position in the constructor but other than gut feeling I can't argument against it so let's keep it for now :).

Reply:
My thoughts were that an initial position was something you'd only want to set when you first create the object, so putting it in the constructor was the only place where this makes sense.